lib/remap-lang/src/object.rs

```diff
@@ -51,4 +51,10 @@ pub trait Object: std::fmt::Debug {
     /// If `compact` is true, after deletion, if an empty object or array is
     /// left behind, it should be removed as well.
     fn remove(&mut self, path: &Path, compact: bool) -> Result<Option<Value>, String>;
+
+    /// An identifier for this object so Remap can identify what type of data it
+    /// is working with.
+    ///
+    /// In vector this will be either `log` or `metric`.
+    fn identifier(&self) -> &'static str;
```

Comment:
Would it make sense to have this return an enum? I expect this was likely done as a string to allow for extensibility. However, since the `is_log` and `is_metric` functions are part of the remap core, you're already hard-coding the three types into the core.

Reply:
All functions are optional. We load them all in through Vector. VRL itself comes without any functions at the moment.

Reply:
> I expect this was likely done as a string to allow for extensibility. 

Precisely. VRL doesn't have to be tied to Vector and `Object` is a core trait of the language. If a third party wanted to use VRL I don't want them to be limited to an Enum we provide. 

I agree, it does feel a bit incongruous having `is_log` and `is_metric` there. 

To make it more generic we could just have a function `object_type()` that returns the identifier. Conditions could then specify `object_type() == "metric"` rather than `is_metric()`. I'm not so keen on that though.

In time perhaps we may split out the functions into Vector specific and generic VRL functions, that would be a non-breaking change we can consider for later.